src/lib/blocs/remark/remark_bloc.dart

```diff
@@ -279,6 +284,143 @@ class RemarkCubit extends Cubit<RemarkState> {
                 .toList())));
   }
 
+  /// Create an empty grading table
+  void initGradingTable() {
+    emit(GradingTabledUpdatedState.updated(
+        initial: state, gradingTable: state.exam.gradingTable));
+  }
+
+  /// Add a new lower bound to the existing grading table
+  void addGradingTableBound() {
+    final copy = state.exam.gradingTable.valueCopy();
+    // Insert empty grading table bound
+    copy.lowerBounds.add(GradingTableLowerBound.empty);
+    emit(GradingTabledUpdatedState.updated(initial: state, gradingTable: copy));
+  }
+
+  /// Change the points on a lower bound in the existing grading table
+  void changeGradingTableBoundPoints(int i, double points) {
+    final copy = state.exam.gradingTable.valueCopy();
+
+    final adjustedLowerBound = copy.lowerBounds[i].copyWith(points: points);
+    // remove old bound
+    copy.lowerBounds.removeAt(i);
+
+    final maxPoints =
+        state.exam.tasks.fold<double>(0.0, (p, c) => p + c.maxPoints);
+    points = math.min(points, maxPoints);
+
+    // insert updated bound at same index
+    copy.lowerBounds.insert(i, adjustedLowerBound);
+
+    // Ensure lower bound constraint
+    for (int j = 0; j < copy.lowerBounds.length; j++) {
+      final lb = state.exam.gradingTable.lowerBounds[j];
+      if (j < i && lb.points < points || j > i && lb.points > points) {
+        log("Adjusting lower bound in grading table");
+
+        final nextLb = lb.copyWith(points: points);
+        // remove old bound
+        copy.lowerBounds.removeAt(j);
+
+        // insert updated bound at same index
+        copy.lowerBounds.insert(j, nextLb);
+      }
+    }
+    emit(GradingTabledUpdatedState.updated(initial: state, gradingTable: copy));
+  }
+
+  /// Change the grade descriptor on a lower bound in the existing grading table
+  void changeGradingTableBoundGrade(int i, String grade) {
+    final copy = state.exam.gradingTable.valueCopy();
+    final adjustedLowerBound = copy.lowerBounds[i].copyWith(grade: grade);
+    // remove old bound
+    copy.lowerBounds.removeAt(i);
+
+    // insert updated bound at same index
+    copy.lowerBounds.insert(i, adjustedLowerBound);
+    emit(GradingTabledUpdatedState.updated(initial: state, gradingTable: copy));
+  }
+
+  /// Change the grading table to the default layout
+  /// The two standard german grading schemes are available as presets
+  /// Source: https://de.wikipedia.org/wiki/Vorlage:Punktesystem_der_gymnasialen_Oberstufe
+  void getDefaultGradingTable(int low, int high) {
```

Comment:
Das hier eventuell in eine Helper-Klasse, sonst blähen wir weiter den RemarkCubit auf.